pkg/exec_unit/plugin_assets.go

```diff
@@ -75,15 +83,49 @@ type assetPathMatcher struct {
 	err     error
 }
 
+func NewAssetPathMatcher(include []string, exclude []string, filePath string) (assetPathMatcher, error) {
+	matcher := assetPathMatcher{}
+	filePath = filepath.ToSlash(filePath)
+	for _, pattern := range include {
+		newPath, err := modifyPathIfRelative(pattern, filePath)
+		if err != nil {
+			return matcher, err
+		}
+		matcher.include = append(matcher.include, newPath)
+	}
+
+	for _, pattern := range exclude {
+		newPath, err := modifyPathIfRelative(pattern, filePath)
+		if err != nil {
+			return matcher, err
+		}
+		matcher.exclude = append(matcher.exclude, newPath)
+	}
+	return matcher, nil
+
+}
+
+func modifyPathIfRelative(path string, currentPath string) (string, error) {
+	if filepath.IsAbs(path) {
+		return strings.TrimPrefix(path, "/"), nil
+	}
+	relPath, err := filepath.Rel(filepath.Dir("."), filepath.Join(filepath.Dir(currentPath), path))
```

Comment:
nit: I think the `Rel` might be redundant since the currentPath is always an input file path (where the annotation was specified) which is always relative to '.'. I'm not sure why it was there originally.

Reply:
Your right, just pushed up a change which makes the logic much simpler